qa/rpc-tests/test_framework/test_framework.py

```diff
@@ -183,12 +179,41 @@ def main(self):
                     from collections import deque
                     print("".join(deque(open(f), MAX_LINES_TO_PRINT)))
         if success:
-            print("Tests successful")
+            self.log.info("Tests successful")
             sys.exit(0)
         else:
-            print("Failed")
+            self.log.warning("Failed")
             sys.exit(1)
 
+    def start_logging(self):
+        # Add logger and logging handlers
+        self.log = logging.getLogger('TestFramework')
+        self.log.setLevel(logging.DEBUG)
+        # Create file handler to log all messages
+        fh = logging.FileHandler(self.options.tmpdir + '/test_framework.log')
+        fh.setLevel(logging.DEBUG)
+        # Create console handler to log messages to stderr. By default this logs only error messages, but can be configured with --loglevel.
+        ch = logging.StreamHandler()
+        console_log_level = getattr(logging, self.options.loglevel.upper(), None)
```

Comment:
Calling getattr doesn't seem to be necessary, the documentation says setLevel already accepts a string: 
https://docs.python.org/3/library/logging.html#logging.Handler.setLevel. Maybe upper() is still useful though.